src/GameData.ts

```diff
@@ -0,0 +1,320 @@
+        if (nameB.startsWith(search) && !nameA.startsWith(search)) return 1;
+
+        if (nameA.includes(search) && !nameB.includes(search)) return -1;
+        if (nameB.includes(search) && !nameA.includes(search)) return 1;
+
+        if (a.item.key && b.item.key) {
+            const keyA = a.item.key!;
+            const keyB = b.item.key!;
+
+            if (keyA === search) return -1;
+            if (keyB === search) return 1;
+
+            if (keyA.startsWith(search) && !keyB.startsWith(search)) return -1;
+            if (keyB.startsWith(search) && !keyA.startsWith(search)) return 1;
+
+            if (keyA.includes(search) && !keyB.includes(search)) return -1;
+            if (keyB.includes(search) && !keyA.includes(search)) return 1;
+        }
+
+        const idA = a.item.id.toString();
+        const idB = a.item.id.toString();
+
+        if (idA === search) return -1;
+        if (idB === search) return 1;
+
+        if (idA.startsWith(search) && !idB.startsWith(search)) return -1;
+        if (idB.startsWith(search) && !idA.startsWith(search)) return 1;
+
+        if (idA.includes(search) && !idB.includes(search)) return -1;
+        if (idB.includes(search) && !idA.includes(search)) return 1;
+
+        if (a.score < b.score) return -1;
+        if (b.score < a.score) return 1;
+
+        return 0;
+    }
+
+    public findItem(search: string): string | any {
+        if (!search) {
+            return `There are currently ${this.itemData.length} items in my lookup data!`;
+        }
+
+        let searchResult: SearchType[];
+        if (search.match(/^\d+$/)) {
+            searchResult = this.itemData.map(i => ({ item: i, score: levenshteinDistance(search, i.id.toString()) }));
+        } else {
+            searchResult = this.itemData.map(i => ({ item: i, score: levenshteinDistance(search, i.name.toLowerCase()) }));
+        }
+
+        searchResult = searchResult.sort((a, b) => this.sortSearch(search, a, b)).slice(0, this.sharedSettings.lookup.maxGuessCount);
+
+        // no exact match, so give alternatives
+        if (searchResult[0].score !== 0) {
+            if (searchResult.length > 0) {
+                let response = "Too many results returned for that search, did you mean one of the options below?\n```";
+                response += searchResult.map(x => `${x.item.name} (Alternate terms: ${x.item.id})`).join("\n");
+                response += "```";
+                return response;
+            } else {
+                return "Unable to find any data for that search term.";
+            }
+        }
+
+        return searchResult.map(x => x.item as ItemData)
+            .map(x => ({
+                id: x.id,
+                name: x.name,
+                combineCost: x.price,
+                cost: x.priceTotal,
+                from: x.from.length > 0 ? x.from : null,
+                to: x.to.length > 0 ? x.to : null,
+            })).slice(0, 1);
```

Comment:
Should probably make `from` and `to` empty arrays, since they will be removed if they are null. IMO it makes more sense to show them as empty than to not show them entirely, but feel free to correct me.